src/sugar3/activity/bundlebuilder.py

```diff
@@ -345,6 +411,61 @@ def _install_desktop_file(self, prefix, activity_path):
         with open(path, 'w') as f:
             cp.write(f)
 
+    def _generate_appdata(self, prefix, activity_path):
+        info = ConfigParser()
+        info.read(os.path.join(activity_path, 'activity', 'activity.info'))
+
+        required_fields = ['metadata_license', 'license', 'name', 'icon',
+                           'description']
+        for name in required_fields:
+            if not info.has_option('Activity', name):
+                print('[WARNING] Activity needs more metadata for AppStream '
+                      'file')
+                print('  Without an AppStream file, the activity will NOT '
+                      'show in software stores!')
+                print('  Please `pydoc sugar3.activity.bundlebuilder` for'
+                      'more info')
+                return
+
+        # See https://www.freedesktop.org/software/appstream/docs/
+        root = ET.Element('component', type='desktop')
+        ET.SubElement(root, 'project_group').text = 'Sugar'
+        ET.SubElement(root, 'id').text = \
+            self.config.bundle_id + '.activity.desktop'
+        desc = ET.fromstring('<description>{}</description>'.format(
+            info.get('Activity', 'description')))
+        root.append(desc)
+
+        copy_pairs = [('metadata_license', 'metadata_license'),
+                      ('license', 'project_license'),
+                      ('summary', 'summary'),
+                      ('name', 'name')]
+        for key, ename in copy_pairs:
+            ET.SubElement(root, ename).text = info.get('Activity', key)
+
+        if info.has_option('Activity', 'screenshots'):
+            screenshots = info.get('Activity', 'screenshots').split()
+            ss_root = ET.SubElement(root, 'screenshots')
+            for i, screenshot in enumerate(screenshots):
+                e = ET.SubElement(ss_root, 'screenshot')
+                if i == 0:
+                    e.set('type', 'default')
+                ET.SubElement(e, 'image').text = screenshot
+
+        if info.has_option('Activity', 'url'):
+            ET.SubElement(root, 'url', type='homepage').text = \
+                info.get('Activity', 'url')
+        if info.has_option('Activity', 'repository_url'):
+            ET.SubElement(root, 'url', type='repository').text = \
```

Comment:
A repository URL type does not exist...
But you might want to add one for the bugtracker.
See https://www.freedesktop.org/software/appstream/docs/chap-Metadata.html#tag-url for details on `<url/>`